src/runtime/storageNG/reference-mode-store.ts

```diff
@@ -498,10 +500,19 @@ export class ReferenceModeStore<Entity extends SerializedEntity, S extends Dicti
     entityCRDT.merge(model);
     const data = entityCRDT.getParticleView();
     for (const [key, value] of Object.entries(data.singletons)) {
-      entity.rawData[key] = value;
+      // For primitives, only the value property of the Referenceable should be included in rawData
+      if (value != undefined && value['value'] != undefined) {
+        entity.rawData[key] = value['value'];
+      } else {
+        entity.rawData[key] = value;
+      }
     }
     for (const [key, value] of Object.entries(data.collections)) {
-      entity.rawData[key] = value;
+      if (value != undefined && value['value'] != undefined) {
```

Comment:
`!== undefined` throughout

Reply:
What is the rule of thumb on when to use null or undefined or ==/!= to encapsulate both? In this case `value` can be `null`, I don't think it can be `undefined` but if it can then the following check would cause an error. I've left it as `value !== null && ...`, is this the right call?

Reply:
You might want to fact check me on this, but from what I remember:

`=== null` only matches null
`=== undefined` only matches undefined
`== null` matches both null and undefined
`== undefined` matches both null and undefined

Usually (not always though), null and undefined are basically treated the same way, so if your variable is nullable you would just check against null, and would catch undefined too. undefined is more of a special case. It usually means a variable has not been assigned a value yet, or when looking up keys in an object/map to say that there's no such element.

I would say a rough rule of thumb is to always use `== null` (or `!= null`) when you want to deal with nullable things in general, and use `=== undefined` (or `!== undefined`) when you specifically know you're dealing with the `undefined` value. That's generally what I've seen used in JS code, but it's by no means a strict rule.

In your specific case, when I saw `value['id'] == undefined` I assumed this was asking "is the `'id'` property defined on the `value` object?" If it's defined, it will return _something_. If it's not defined, it will return `undefined` (that's the return value from a failed lookup on an object's property), hence the check should really be for `=== undefined`.

Reply:
(Another rule of thumb is to prefer to use `===` in JavaScript whenever you can, because there are so many gotchas... I would say that `== null` is one of the common exceptions, for when you want to match both null and undefined)